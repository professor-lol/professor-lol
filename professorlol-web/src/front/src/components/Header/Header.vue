<template>
  <header
    :class="classList">
    <slot>Header</slot>
  </header>
</template>

<script>
export default {
  name: 'AppHeader',
  props: {
    fixed: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    classList () {
      return [
        'app-header',
        'navbar'
      ]
    }
  },
  mounted: function () {
    this.isFixed(this.fixed)
  },
  methods: {
    isFixed (fixed) {
      fixed ? document.body.classList.add('header-fixed') : document.body.classList.remove('header-fixed')
      return fixed
    }
  }
}
</script>
