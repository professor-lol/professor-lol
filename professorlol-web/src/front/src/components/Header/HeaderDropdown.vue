<template>
  <b-nav-item-dropdown :right="right" :no-caret="noCaret">
    <template slot="button-content">
      <slot name="header">
        &#10068;
      </slot>
    </template>
    <slot name="dropdown">
      <div :style="{ right: 'auto', height: '200px' }">
        <span class="text-center">dropdown</span>
      </div>
    </slot>
  </b-nav-item-dropdown>
</template>
<script>
export default {
  name: 'HeaderDropdown',
  props: {
    right: {
      type: Boolean,
      default: false
    },
    noCaret: {
      type: Boolean,
      default: false
    }
  }
}
</script>
