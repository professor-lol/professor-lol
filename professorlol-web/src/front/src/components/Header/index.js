import Header from './Header'
import HeaderDropdown from './HeaderDropdown'

export {
  Header,
  HeaderDropdown
}
