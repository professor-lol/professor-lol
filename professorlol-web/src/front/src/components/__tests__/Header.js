import { mount } from "@vue/test-utils";
import Header from "../Header/Header";

describe("Header.vue", () => {
  // Inspect the raw component options
  it('has isFixed method', () => {
    expect(typeof Header.methods.isFixed).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(Header)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.element.textContent).toEqual('Header')
    expect(wrapper.classes()).toContain('app-header')
    expect(wrapper.classes()).toContain('navbar')
  })
});
