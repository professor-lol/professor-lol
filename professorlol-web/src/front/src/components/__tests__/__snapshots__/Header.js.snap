// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue renders correctly 1`] = `
<header
  class="app-header navbar"
>
  Header
</header>
`;
